packages/page-staking/src/Nomination/AccountSelector.tsx

```diff
@@ -0,0 +1,78 @@
+// Copyright 2017-2020 @polkadot/app-123code authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { InputAddress} from '@polkadot/react-components';
+import { useBalanceClear } from '@polkadot/app-staking/Nomination/useBalance';
+import { useApi} from '@polkadot/react-hooks';
+
+interface Props {
+  value?: string | null;
+  className?: string;
+  title?: string;
+  onChange: (accountId: string | null) => void;
+  stepsState: string[];
+  setStepsState: (stepsState: string[]) => void; // Dispatch<SetStateAction<string>>
+}
+
+function AccountSelector ({ className, onChange, title, stepsState, setStepsState, value }: Props): React.ReactElement<Props> {
+  const [accountId, setAccountId] = useState<string | null>(null);
+  const balance = useBalanceClear(accountId);
+  const api = useApi();
+  const existentialDeposit = api.api.consts.balances.existentialDeposit;
+
+  useEffect((): void => {
+      if (accountId) {
+        onChange(accountId)
+      }
+  },[accountId]);
+
+  useEffect(() => {
+    const newStepsState = [...stepsState];
```

Comment:
The hooks linter will complain about this. Not exhaustive. If you want the previous state, the best-practice is always to do -

```js
setStepsState((state) => {
  // do stuff here...

  return newStepsState;
});
```